src/term/mod.rs

```diff
@@ -436,18 +438,53 @@ impl Term {
             size_info: size,
             empty_cell: template,
             custom_cursor_colors: config.custom_cursor_colors(),
+            semantic_escape_chars: config.semantic_escape_chars().clone(),
         }
     }
 
     pub fn update_config(&mut self, config: &Config) {
-        self.custom_cursor_colors = config.custom_cursor_colors()
+        self.custom_cursor_colors = config.custom_cursor_colors();
+        self.semantic_escape_chars = config.semantic_escape_chars().clone();
     }
 
     #[inline]
     pub fn needs_draw(&self) -> bool {
         self.dirty
     }
 
+    pub fn semantic_selection(&self, selection: &mut Selection, point: Point) {
```

Comment:
Seems like this doesn't handle triple clicking at all. Is that intended to use a separate function?